tests/rbd_mirror/test_configure_mirror_snapshot.py

```diff
@@ -0,0 +1,65 @@
+from tests.rbd_mirror import rbd_mirror_utils as rbdmirror
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(**kw):
+    """
+    Creates user specified Pool and enables snapshot based Mirroring
+    on each images in that pool.
+    Args:
+        **kw: test data
+        Example::
+            config:
+                imagesize: 2G
+    Returns:
+        0 - if test case pass
+        1 - if test case fails
+    """
+    try:
+        log.info("Configuring RBD mirroring with snapshot based")
+
+        config = kw.get("config")
+        poolname = config.get("poolname")
+        mode = "image"
+        mirrormode = "snapshot"
+        imagesize = config.get("imagesize", "1G")
+        number_of_images = config.get("image-count", 10)
+
+        mirror1 = rbdmirror.RbdMirror(
+            kw.get("ceph_cluster_dict").get("ceph-rbd1"), config
+        )
+        mirror2 = rbdmirror.RbdMirror(
+            kw.get("ceph_cluster_dict").get("ceph-rbd2"), config
+        )
+
+        imagename = "tier_2_rbd_mirror_image" + mirror1.random_string()
```

Comment:
Please generalize :(
Should we be creating new file with almost same code for tier 3 testcases ?

Reply:
As it's needed 3 cluster scenario, and some user-specified image creation hence for safer side keep it as separate module.

Reply:
(there is no co-relation between how module is working and your answer)

Reply:
sorry got confused, changed the image name and updated the module accordingly `reusability and optimzation` perspective.